src/overlays/actors/ovl_En_Wdhand/z_en_wdhand.c

```diff
@@ -105,63 +175,679 @@ static DamageTable D_80AF64F4 = {
+
+        if (this->timer == 20 * i) {
+            if (i != 0) {
+                this->limbRotations[i].y = this->limbRotations[i - 1].y + (s32)Rand_CenteredFloat(16384.0f);
+                this->limbRotations[i].y -= 0x8000;
+            } else {
+                this->limbRotations[0].y += (s16)(s32)Rand_CenteredFloat(16384.0f);
+            }
+            this->limbRotations[i].z = Rand_S16Offset(0x1800, 0x1000);
+        }
+    }
+
+    if (this->timer == 0) {
+        this->timer = 80;
+    }
+
+    if (!(player->stateFlags2 & PLAYER_STATE2_80) && (this->actor.xyzDistToPlayerSq < SQ(120.75f))) {
+        EnWdhand_GetRelativeRotationsToPlayerRightFoot(this, player, &pitchToPlayerRightFoot, &yawToPlayerRightFoot);
+        if (ABS_ALT(pitchToPlayerRightFoot) <= 0x4000) {
+            EnWdhand_SetupLungeForPlayer(this);
+        }
+    }
+}
+
+void EnWdhand_SetupLungeForPlayer(EnWdhand* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gDexihandOpenAnim, -5.0f);
+    this->timer = 5;
+    this->actionFunc = EnWdhand_LungeForPlayer;
+}
+
+void EnWdhand_LungeForPlayer(EnWdhand* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s32 allStepsDone = true;
+    s32 i;
+    s16 pitchToPlayerRightFoot;
+    s16 yawToPlayerRightFoot;
+
+    SkelAnime_Update(&this->skelAnime);
+
+    // Only activate the collider and grab check after a short delay
+    if (this->timer > 0) {
+        this->timer--;
+        if (this->timer == 0) {
+            this->collider.base.atFlags |= AT_ON;
+        }
+        return;
+    }
+
+    Math_StepToF(&this->globalLimbScaleFactor, 1.5f, 0.05f);
+    EnWdhand_GetRelativeRotationsToPlayerRightFoot(this, player, &pitchToPlayerRightFoot, &yawToPlayerRightFoot);
+
+    for (i = 0; i < ARRAY_COUNT(this->limbRotations); i++) {
+        s16 yawDiff = yawToPlayerRightFoot - this->limbRotations[i].y;
+
+        if (ABS_ALT(yawDiff) < 0x4000) {
+            allStepsDone &=
+                Math_ScaledStepToS(&this->limbRotations[i].x, pitchToPlayerRightFoot, (i * 0.1f + 1.0f) * 1920.0f);
+            allStepsDone &= Math_ScaledStepToS(&this->limbRotations[i].y, yawToPlayerRightFoot, 0x800);
+        } else {
+            allStepsDone &=
+                Math_ScaledStepToS(&this->limbRotations[i].x, -pitchToPlayerRightFoot, (i * 0.1f + 1.0f) * 1920.0f);
+            allStepsDone &= Math_ScaledStepToS(&this->limbRotations[i].y, yawToPlayerRightFoot + 0x8000, 0x800);
+        }
+    }
+
+    if ((this->collider.base.atFlags & AT_HIT) && play->grabPlayer(play, player)) {
+        // Touched the player, grab
+        EnWdhand_SetupGrabbedPlayer(this, play);
+    } else if (allStepsDone) {
+        // Failed to grab player in time
+        EnWdhand_SetupFailedGrabPlayer(this);
+    }
+
+    if (this->collider.base.atFlags & AT_HIT) {
+        this->collider.base.atFlags &= ~AT_HIT;
+    }
+}
+
+void EnWdhand_SetupFailedGrabPlayer(EnWdhand* this) {
```

Comment:
Nit: Maybe do `SetupFailedToGrabPlayer`/`FailedToGrabPlayer` for grammatical correctness?